pkg/proxy/userspace/proxier.go

```diff
@@ -289,6 +324,44 @@ func CleanupLeftovers(ipt iptables.Interface) (encounteredError bool) {
 
 // Sync is called to immediately synchronize the proxier state to iptables
 func (proxier *Proxier) Sync() {
+	proxier.syncRunner.Run()
+}
+
+func (proxier *Proxier) isInitialized() bool {
+	proxier.mu.Lock()
+	defer proxier.mu.Unlock()
+	return proxier.servicesSynced && proxier.endpointsSynced
+}
+
+func (proxier *Proxier) syncProxyRules() {
+	start := time.Now()
+	defer func() {
```

Comment:
maybe this should go after the fast-return if not initialized?

Reply:
yep